src/service_daemon.rs

```diff
@@ -482,10 +482,13 @@ impl ServiceDaemon {
     /// 5. process retransmissions if any.
     fn run(mut zc: Zeroconf, receiver: Receiver<Command>) -> Option<Command> {
         // Add the daemon's signal socket to the poller.
-        if let Err(e) = zc.poller.add(
-            &zc.signal_sock,
-            polling::Event::readable(SIGNAL_SOCK_EVENT_KEY),
-        ) {
+        #[allow(unsafe_code)]
```

Comment:
Thanks for your PR! I want to keep this library safe code only, unless we really have to break it. 

Let me take a look at our options.